java/src/main/native/ai_onnxruntime_OnnxSparseTensor.c

```diff
@@ -0,0 +1,528 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+#include <jni.h>
+#include <math.h>
+#include <stdlib.h>
+#include "onnxruntime/core/session/onnxruntime_c_api.h"
+#include "OrtJniUtil.h"
+#include "ai_onnxruntime_OnnxSparseTensor.h"
+
+/*
+ * Class:     ai_onnxruntime_OnnxSparseTensor
+ * Method:    getIndicesBuffer
+ * Signature: (JJ)Ljava/nio/ByteBuffer;
+ */
+JNIEXPORT jobject JNICALL Java_ai_onnxruntime_OnnxSparseTensor_getIndicesBuffer
+    (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
+  (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*) apiHandle;
+  OrtSparseFormat format;
+  OrtErrorCode code = checkOrtStatus(jniEnv, api, api->GetSparseTensorFormat((OrtValue*) handle, &format));
+  if (code != ORT_OK) {
+    return NULL;
+  }
+  enum OrtSparseIndicesFormat indicesFormat;
+  switch (format) {
+    case ORT_SPARSE_COO:
+        indicesFormat = ORT_SPARSE_COO_INDICES;
+        break;
+    case ORT_SPARSE_CSRC:
+        indicesFormat = ORT_SPARSE_CSR_OUTER_INDICES;
+        break;
+    case ORT_SPARSE_BLOCK_SPARSE:
+        indicesFormat = ORT_SPARSE_BLOCK_SPARSE_INDICES;
+        break;
+    case ORT_SPARSE_UNDEFINED: {
+        throwOrtException(jniEnv, convertErrorCode(ORT_NOT_IMPLEMENTED), "Sparse format is ORT_SPARSE_UNDEFINED, cannot get indices");
+        return NULL;
+    }
+  }
+
+  OrtTensorTypeAndShapeInfo* info = NULL;
+  code = checkOrtStatus(jniEnv, api, api->GetSparseTensorIndicesTypeShape((OrtValue*) handle, indicesFormat, &info));
+  if (code != ORT_OK) {
+    return NULL;
+  }
+  size_t arrSize = 0;
+  code = checkOrtStatus(jniEnv, api, api->GetTensorShapeElementCount(info, &arrSize));
+  if (code != ORT_OK) {
+    api->ReleaseTensorTypeAndShapeInfo(info);
+    return NULL;
+  }
+  ONNXTensorElementDataType onnxTypeEnum;
+  code = checkOrtStatus(jniEnv, api, api->GetTensorElementType(info, &onnxTypeEnum));
+  api->ReleaseTensorTypeAndShapeInfo(info);
+  if (code != ORT_OK) {
+    return NULL;
+  }
+
+  size_t typeSize = onnxTypeSize(onnxTypeEnum);
+  size_t sizeBytes = arrSize * typeSize;
+
+  uint8_t* arr = NULL;
+  size_t indices_size = 0;
+  code = checkOrtStatus(jniEnv, api, api->GetSparseTensorIndices((OrtValue*)handle, indicesFormat, &indices_size, (const void**)&arr));
```

Comment:
>(OrtValue* [](http://example.com/codeflow?start=65&length=10)

I believe the api takes in const OrtValue*. Same is above for non-mutating interfaces